lisa/lisa-core/src/main/java/it/unive/lisa/analysis/string/Suffix.java

```diff
@@ -0,0 +1,191 @@
+package it.unive.lisa.analysis.string;
+
+import it.unive.lisa.analysis.Lattice;
+import it.unive.lisa.analysis.SemanticDomain.Satisfiability;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.analysis.nonrelational.value.BaseNonRelationalValueDomain;
+import it.unive.lisa.analysis.representation.DomainRepresentation;
+import it.unive.lisa.analysis.representation.StringRepresentation;
+import it.unive.lisa.program.cfg.ProgramPoint;
+import it.unive.lisa.symbolic.value.Constant;
+import it.unive.lisa.symbolic.value.operator.binary.*;
+import it.unive.lisa.symbolic.value.operator.ternary.TernaryOperator;
+import it.unive.lisa.symbolic.value.operator.unary.UnaryOperator;
+import java.util.Objects;
+
+/**
+ * The suffix string abstract domain.
+ *
+ * @author <a href="mailto:vincenzo.arceri@unipr.it">Vincenzo Arceri</a>
+ * @author <a href="mailto:sergiosalvatore.evola@studenti.unipr.it">Sergio
+ *             Salvatore Evola</a>
+ * 
+ * @see <a href=
+ *          "https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34">
+ *          https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34</a>
+ */
+public class Suffix extends BaseNonRelationalValueDomain<Suffix> {
+
+	private final static Suffix TOP = new Suffix();
+	private final static Suffix BOTTOM = new Suffix(null);
+	private final String suffix;
+
+	/**
+	 * Builds the top suffix abstract element.
+	 */
+	public Suffix() {
+		this("");
+	}
+
+	/**
+	 * Builds a suffix abstract element.
+	 *
+	 * @param suffix the suffix
+	 */
+	public Suffix(String suffix) {
+		this.suffix = suffix;
+	}
+
+	@Override
+	public Suffix lubAux(Suffix other) throws SemanticException {
+		String otherSuffix = other.getSuffix();
```

Comment:
Valid here and in the other classes: as those getters just return the field, it's faster to access the field directly instead of calling the method